lib/Analysis/Allocation.cpp

```diff
@@ -39,8 +39,10 @@ getCvtOrder(Attribute srcLayout, Attribute dstLayout) {
   auto srcDotLayout = srcLayout.dyn_cast<DotOperandEncodingAttr>();
   auto dstMmaLayout = dstLayout.dyn_cast<MmaEncodingAttr>();
   auto dstDotLayout = dstLayout.dyn_cast<DotOperandEncodingAttr>();
-  assert(!(srcMmaLayout && dstMmaLayout) &&
-         "Unexpected mma -> mma layout conversion");
+
+  assert(!(srcMmaLayout && dstMmaLayout && srcMmaLayout.isVolta()) &&
```

Comment:
How about hopper?

Reply:
Good question -- for some reason, I assumed mma -> mma layout conversion was supported on Hopper. But this is not the case.

Conversion of  `MmaLayout((3, 0), [1, 4], [1, 1], [1, 1], [0, 1], [16, 8, 16])` to `MmaLayout((3, 0), [4, 1], [1, 1], [1, 1], [0, 1], [16, 8, 16])` triggers this assertion:

https://github.com/openai/triton/blob/bd95d2d0de3a58c182192bb52e520550db8a48d0/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp#L791

I'll investigate. The main thing that is confusing is, the error happens inside a code-path for blocked to shared layouts.